common/persistence/elasticsearch/visibility_store.go

```diff
@@ -1135,3 +1153,55 @@ func statusProcessFunc(_ *fastjson.Object, _ string, value *fastjson.Value) erro
 			return nil
 		})
 }
+func durationKeyFilter(key string) bool {
+	return key == searchattribute.ExecutionDuration
+}
+
+func durationProcessFunc(_ *fastjson.Object, _ string, value *fastjson.Value) error {
+	return processAllValuesForKey(
+		value,
+		func(key string) bool {
+			_, ok := rangeKeys[key]
+			return ok
+		},
+		func(obj *fastjson.Object, key string, v *fastjson.Value) error {
+			durationStr := string(v.GetStringBytes())
+
+			// To support durations passed as golang durations such as "300ms", "-1.5h" or "2h45m".
+			// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+			if duration, err := time.ParseDuration(durationStr); err == nil {
+				obj.Set(key, fastjson.MustParse(strconv.FormatInt(duration.Nanoseconds(), 10)))
+				return nil
+			}
+
+			// To support "hh:mm:ss" durations.
+			durationNanos, err := parseHHMMSSDuration(durationStr)
+			if errors.Is(err, ErrInvalidDuration) {
+				return err
+			}
+			if err == nil {
+				obj.Set(key, fastjson.MustParse(strconv.FormatInt(durationNanos, 10)))
+			}
+
+			return nil
+		})
+}
+
+func parseHHMMSSDuration(d string) (int64, error) {
```

Comment:
is this format commonly used? I would suggest not to support this format. 10:30:00 looks like 10:30am.

Reply:
I used to use it in C# and it seems to be popular.